crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -56,84 +138,265 @@ impl TryFrom<&JpmorganRouterData<&PaymentsAuthorizeRouterData>> for JpmorganPaym
+pub trait FromResponseStatus {
+    fn from_response_status(transaction_state: String) -> Self;
+}
+
+impl FromResponseStatus for common_enums::AttemptStatus {
+    fn from_response_status(transaction_state: String) -> Self {
+        match transaction_state.as_str() {
+            "Success" => Self::Voided,
+            _ => Self::Failure,
+        }
+    }
 }
 
-impl From<JpmorganPaymentStatus> for common_enums::AttemptStatus {
-    fn from(item: JpmorganPaymentStatus) -> Self {
+impl From<JpmorganTransactionStatus> for common_enums::AttemptStatus {
+    fn from(item: JpmorganTransactionStatus) -> Self {
         match item {
-            JpmorganPaymentStatus::Succeeded => Self::Charged,
-            JpmorganPaymentStatus::Failed => Self::Failure,
-            JpmorganPaymentStatus::Processing => Self::Authorizing,
+            JpmorganTransactionStatus::Success => Self::Charged,
+            JpmorganTransactionStatus::Denied | JpmorganTransactionStatus::Error => Self::Failure,
         }
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
-pub struct JpmorganPaymentsResponse {
-    status: JpmorganPaymentStatus,
-    id: String,
+impl FromTransactionState for common_enums::AttemptStatus {
+    fn from_transaction_state(transaction_state: JpmorganTransactionState) -> Self {
+        match transaction_state {
+            JpmorganTransactionState::Authorized => Self::Authorized,
+            JpmorganTransactionState::Closed => Self::Charged,
+            JpmorganTransactionState::Declined | JpmorganTransactionState::Error => Self::Failure,
+            JpmorganTransactionState::Pending => Self::Pending,
+            JpmorganTransactionState::Voided => Self::Voided,
+        }
+    }
 }
 
 impl<F, T> TryFrom<ResponseRouterData<F, JpmorganPaymentsResponse, T, PaymentsResponseData>>
     for RouterData<F, T, PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
+
     fn try_from(
         item: ResponseRouterData<F, JpmorganPaymentsResponse, T, PaymentsResponseData>,
     ) -> Result<Self, Self::Error> {
+        let transaction_state = match item.response.transaction_state {
+            JpmorganTransactionState::Closed => match item.response.capture_method.as_deref() {
+                Some("NOW") => JpmorganTransactionState::Closed,
+                _ => JpmorganTransactionState::Authorized,
+            },
+            JpmorganTransactionState::Authorized => JpmorganTransactionState::Authorized,
+            JpmorganTransactionState::Voided => JpmorganTransactionState::Voided,
+            JpmorganTransactionState::Pending => JpmorganTransactionState::Pending,
+            JpmorganTransactionState::Declined => JpmorganTransactionState::Declined,
+            JpmorganTransactionState::Error => JpmorganTransactionState::Error,
+        };
+
+        let status = common_enums::AttemptStatus::from_transaction_state(transaction_state);
+
```

Comment:
What is happening here?

Reply:
So, `Closed` TransactionStatus has two meanings, if in the res we have captureMethod as `NOW`, that means TransactionStatus will be `Closed`, and if its `MANUAL`, that means TransactionStatus will be `AUTHORIZED`